src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy

```diff
@@ -155,6 +156,40 @@ class GitVersionPluginTests extends Specification {
         buildResult.output.contains(":printVersion\n1.0.0\n")
     }
 
+    def 'git describe when annotated tag is present with merge commit' () {
+        given:
+        buildFile << '''
+            plugins {
+                id 'com.palantir.git-version'
+            }
+            version gitVersion()
+        '''.stripIndent()
+        gitIgnoreFile << 'build'
+
+        // create repository with a single commit tagged as 1.0.0
+        Git git = Git.init().setDirectory(projectDir).call();
+        git.add().addFilepattern('.').call()
+        git.commit().setMessage('initial commit').call()
+        git.tag().setAnnotated(true).setMessage('1.0.0').setName('1.0.0').call()
+
+        // create a new branch called "hotfix" that has a single commit and is tagged with "1.0.0-hotfix"
+        String master = git.getRepository().getFullBranch();
+        Ref hotfixBranch = git.branchCreate().setName("hotfix").call()
+        git.checkout().setName(hotfixBranch.getName()).call()
+        git.commit().setMessage("hot fix for issue").call()
+        git.tag().setAnnotated(true).setMessage('1.0.0-hotfix').setName('1.0.0-hotfix').call()
+
+        // switch back to main branch and merge hotfix branch into main branch
+        git.checkout().setName(master).call()
+        git.merge().include( git.getRepository().getRef("hotfix")).setFastForward(MergeCommand.FastForwardMode.NO_FF).setMessage("merge commit").call()
+
+        when:
+        BuildResult buildResult = with('printVersion').build()
+
+        then:
+        buildResult.output =~ ":printVersion\n1.0.0-1-g[a-z0-9]{7}\n"
+    }
```

Comment:
I'd like to see the same test but also tagging a 2.0.0 after the hotfix merges to ensure we see that come up.